src/TextInput.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import styled from '@emotion/styled';
+
+// TextInput is a generic text input component.
+
+interface IProps extends React.InputHTMLAttributes<HTMLInputElement> {
+  error?: string;
+  label?: string;
+}
+
+const Error = styled.p`
+  color: red;
+  font-size: .8rem;
+  margin: .2rem 0 0;
+  text-align: right;
+`;
+
+const Label = styled.label`
+  font-size: 0.9rem;
+  font-weight: 600;
+  display: block;
+  margin-bottom: .2rem;
+`;
+
+const Input = styled.input`
+  border: solid 1px #bbb;
+  border-radius: 2px;
+  display: block;
+  font-size: 1rem;
+  height: 2.6rem;
+  padding: 0.5rem;
+  width: 100%;
+`;
+
+export const TextInput = ({ error, label, ...rest }: IProps) => (
```

Comment:
What does the ... operator mean here?

Reply:
This is the rest operator in conjunction with object destructuring: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters

In this example the following happens:
- the property `error` inside the object is assigned to the variable `error`
- the property `label` inside the object is assigned to the variable `label`
- every other property, excluding the above 2, is placed in the variable `rest`